src/test/groovy/org/shipkit/internal/gradle/init/tasks/InitShipkitFileTest.groovy

```diff
@@ -0,0 +1,70 @@
+package org.shipkit.internal.gradle.init.tasks
+
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import org.shipkit.internal.notes.vcs.GitOriginRepoProvider
+import spock.lang.Specification
+
+class InitShipkitFileTest extends Specification {
+
+    @Rule TemporaryFolder tmp = new TemporaryFolder()
+
+    def "does not modify shipkit file if it already exists"() {
+        given:
+        def shipkitFile = tmp.newFile()
+        shipkitFile << "foo"
+
+        when:
+        InitShipkitFile.initShipkitFile(shipkitFile, tmp.root)
+
+        then:
+        shipkitFile.text == "foo"
+    }
+
+    def "uses fallback repo if git call fails"() {
+        def shipkitFile = tmp.newFile()
+        def repoProvider = Mock(GitOriginRepoProvider)
```

Comment:
hmmm, I've tried that but the code is subjectively harder to read when some local variables use formal types and some not. Why do you push for formal types for local variables?